src/bio/bio_monitor.c

```diff
@@ -462,6 +477,8 @@ bio_bs_monitor(struct bio_xs_context *ctxt, uint64_t now)
 			ctxt->bxc_tgt_id, rc);
 
 	collect_raw_health_data(dev_health);
+	collect_bs_usage(ctxt->bxc_blobstore->bb_bs,
+			 &dev_health->bdh_health_state);
```

Comment:
Minor: We don't have to call this collect_bs_usage() periodically in bio_bs_monitor(), just call collect_bs_usage() in the query handler function like bio_get_dev_state_internal() should be enough. Also, calling collect_bs_usage() on query handler can get up to date space usage. (This bio_bs_monitor() is called by owner xstream every 60 seconds, so if any pool created/destroyed after last bio_bs_monitor() call, the query could return stale usage information)

Reply:
yes , that sounds like a good change , but maybe I could do that in a follow-on?

Reply:
Since we are in the process of running IO-500, i would rather address this now before landing the PR because we spend time investigating why performance drops every minute.

Reply:
sure